SU-HSDO/suhumsci
# feat: use we_megamenu preprocess hook to add private page icon before private pages in dropdown

# READY FOR REVIEW

## Summary
Currently, if there is a Drupal private page in the `we_megamenu` dropdown, there is no indicator present to make the user aware that the page is private. This PR uses the `we_megamenu` preprocess hook for list items to add a lock icon before the `<a>` element of a private page, matching how private pages are currently indicated in sidebar menus.

## Need Review By (Date)
asap

## Urgency
medium

## Steps to Test
1. Make sure the Mega Menu (`we_megamenu`) is installed, and its corresponding block is placed in the `Menu` region on your local environment. If not, follow the instructions in https://github.com/SU-HSDO/suhumsci/pull/1074 to set up the Mega Menu locally.
2. Be sure to be testing on a desktop view, since Mega Menu is a desktop-only feature.
3. Hover over the 'About' item in the Mega Menu navigation
4. Look for the link item title 'Private Page`
5. Verify that there is now a black lock icon preceding the 'Private Page` link text.

<img width="560" alt="Screen Shot 2022-04-08 at 4 34 57 PM" src="https://user-images.githubusercontent.com/2685062/162526717-c2ca789c-2b3b-4c65-ad6d-a1032d3670b0.png">

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)
